synthesizer/src/vm/mod.rs

```diff
@@ -109,81 +112,71 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
+        // Load the deployments and their block heights from the store.
         for (i, chunk) in deployment_ids.chunks(256).enumerate() {
             debug!(
                 "Loading deployments {}-{} (of {})...",
                 i * 256,
                 ((i + 1) * 256).min(deployment_ids.len()),
                 deployment_ids.len()
             );
-            let deployments = cfg_iter!(chunk)
-                .map(|transaction_id| {
-                    // Load the deployment and its imports.
-                    load_deployment_and_imports(&process, transaction_store, **transaction_id)
-                })
-                .collect::<Result<Vec<_>>>()?;
-
-            for (program_id, deployment) in deployments.iter().flatten() {
-                // Load the deployment if it does not exist in the process yet.
-                if !process.contains_program(program_id) {
-                    process.load_deployment(deployment)?;
+            // Load the deployments and their block heights.
+            let chunk_iter = cfg_iter!(chunk).map(|transaction_id| {
+                // Retrieve the deployment from the transaction ID.
+                let deployment = match transaction_store.get_deployment(transaction_id)? {
+                    Some(deployment) => deployment,
+                    None => bail!("Deployment transaction '{transaction_id}' is not found in storage."),
+                };
+                // Retrieve the height.
+                let height = match block_store
+                    .find_block_hash(transaction_id)?
+                    .map(|hash| block_store.get_block_height(&hash))
+                {
+                    Some(Ok(Some(height))) => height,
+                    _ => {
+                        bail!("Block height for deployment transaction '{transaction_id}' is not found in storage.")
+                    }
+                };
+                // Return the deployment and height.
+                let transaction_id = transaction_id.clone();
+                let local_map = HashMap::from([(*transaction_id, deployment)]);
+                let local_vec = vec![(*transaction_id, height)];
+                Ok((local_map, local_vec))
+            });
+            // Aggregate the chunk.
+            let (local_deployments, local_heights) = cfg_reduce!(
+                chunk_iter,
+                || Ok((HashMap::new(), Vec::new())),
+                |acc: Result<_, anyhow::Error>, local| {
+                    let mut acc = acc?;
+                    let local = local?;
+                    acc.0.extend(local.0);
+                    acc.1.extend(local.1);
+                    Ok(acc)
                 }
-            }
+            )?;
+            // Extend the deployments and heights.
+            deployments.extend(local_deployments);
+            heights.extend(local_heights);
+        }
+
+        // Sort the deployments by their block heights.
+        heights.sort_by(|(_, a), (_, b)| a.cmp(b));
```

Comment:
```suggestion
        heights.sort_unstable_by(|(_, a), (_, b)| a.cmp(b));
```
I'm assuming equal heights (if we can even have duplicate heights here) may be reordered